### PR TITLE
fix the py3status

### PR DESCRIPTION
**Pull request by tardyp**
I merged too early :-}

### CODE REVIEW

**Comment by seankelly** [comment]
Would also be nice if the output got written to stdout/stderr.

File: .travis.yml [hunk 1]
```diff
@@ -130,14 +130,13 @@ script:
   - "[ $TESTS != js -o ${TRAVIS_PULL_REQUEST} != false ] || make frontend_install_tests"
 
   - |
-    TESTS=py3status
     if [ $TESTS = py3status ]; then
         trial  --reporter=summary buildbot.test buildbot_worker.test| tail -n 1 | python -c "
```
Comment (seankelly) [inline .travis.yml]: Maybe:

    trial --reporter=summary buildbot.test buildbot_worker.test | tee py3status
    tail -n 1 py3status | python -c "
    ...

**Comment by rodrigc** [comment]
In .travis.yml, there are a bunch of lines like:

```
 - "[ $TESTS != lint ] || make pylint"
 - "[ $TESTS != lint ] || make flake8"
 - "[ $TESTS != lint ] || isort --check -df `git ls-files |grep '.py$'`"
```

Why isn't the logic like:

```
 - "[ $TESTS = lint ] && make pylint"
 - "[ $TESTS = lint ] && make flake8"
 - "[ $TESTS = lint ] && isort --check -df `git ls-files |grep '.py$'`"
```

That would make the logic a lot more readable.

**Comment by tardyp** [comment]
@rodrigc I agree, I dont know exactly the reason. I guess this is from a best practice in travis doc, but I have no idea why.
in bbtravis, we are using a better DSL imho.

**Comment by tardyp** [comment]
@rodrigc I this that is because if you use the && form, then the shell will not detect the return value in case of error

**Comment by rodrigc** [comment]
What do you care what the error is if you have:

```
[ $TESTS == foo ] && do_foo
```

$TESTS is just a string that you are comparing.

**Comment by tardyp** [comment]
@rodrigc 
if you do 

    - [ $TESTS == foo ] && do_foo

and $TEST is e.g. equal to 'trial', then  [ $TESTS == foo ]  is false, so the return value will not be zero, so this line of shell will be considered failure, and the build will be marked fail :-(

**Comment by rutsky** [comment]
The problem is that Travis support different envs, but doesn't support different scripts for different envs to run.
So different scripts are emulated in such ugly way.

**Comment by rutsky** [comment]
Not "the" problem of `&& vs ||`, but the reason why logic is "define N cools envs, and run single ugly script with N conditions to switch between them".

**Comment by tardyp** [comment]
image

at least this works..

**Comment by rodrigc** [comment]
Are you sure this is working as you want?
If I input the .travis.yml to: https://lint.travis-ci.org, I get:

```
in matrix.allow_failures section: unexpected key TESTS, dropping
```

**Pull request merged**